custom_metrics/privacy.js

```diff
@@ -15,10 +46,264 @@ return JSON.stringify({
+   *
+   * Test site: https://www.rtl.de/
+   */
+  iab_tcf_v2: (() => {
+    let tcData = {
+      present: typeof window.__tcfapi == 'function',
+      data: null,
+      compliant_setup: null,
+    };
+    // description of `__tcfapi`: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-provide-the-api
+    if (tcData.present) {
+      // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+      window.__tcfapi('getTCData', 2, (result, success) => {
+        if (success) {
+          tcData.data = result;
+          tcData.compliant_setup = true;
+        } else {
+          // special case for consentmanager ('CMP settings are used that are not compliant with the IAB TCF')
+          // see warning at the top of https://help.consentmanager.net/books/cmp/page/changes-to-the-iab-cmp-framework-js-api
+          // cf. https://help.consentmanager.net/books/cmp/page/javascript-api
+          // Test site: https://www.pokellector.com/
+          window.__tcfapi('noncompliant_getTCData', 2, (result, success) => {
+            if (success) {
+              tcData.data = result;
+              tcData.compliant_setup = false;
+            }
+          });
+        }
+      });
+    }
+    return tcData;
+  })(),
+
+  /**
+   * IAB US Privacy User Signal Mechanism “USP API”
+   * https://github.com/InteractiveAdvertisingBureau/USPrivacy
+   *
+   * Test site: https://www.nfl.com/
+   */
+  iab_usp: (() => {
+    let uspData = {
+      present: typeof window.__uspapi == 'function',
+      privacy_string: null,
+    };
+    if (uspData.present) {
+      window.__uspapi('getUSPData', 1, (result, success) => {
+        if (success) {
+          uspData.privacy_string = result;
+        }
+      });
+    }
+    return uspData;
+  })(),
+
+  /**
+   * Ads Transparency Spotlight Data Disclosure schema
+   * Only for top frame, can't access child frames (same-origin policy)
+   *
+   * Test site: unknown
+   */
+  ads_transparency_spotlight: (() => {
+    // Check `meta` tag cf. https://github.com/Ads-Transparency-Spotlight/documentation/blob/main/implement.md
+    meta_tag = document.querySelector('meta[name="AdsMetadata"]');
+    let ats = {
+      present: meta_tag != null,
+      ads_metadata: null,
+    };
+    if (ats.present) {
+      ats.ads_metadata = meta_tag.content;
+    }
+    return ats;
+  })(),
+
+  /**
+   * FLoC
+   *
+   * Test site: https://floc.glitch.me/
+   * Test site: https://www.pokellector.com/
+   *
+   * @todo Check function/variable accesses through string searches (wrappers cannot be used, as the metrics are only collected at the end of the test)
+   */
+  floc: {
+    document_interestCohort:
+      findStringInBodies('document.interestCohort') ||
+      findStringInBodies('interestCohort'),
```

Comment:
Are these redundant? Everything matched by the first pattern will also be matched by the second, right?

Reply:
Yes, you're right. Will change it to make sure all of the keywords are part of the specific JS file.

Reply:
done in 3cb05be19658a80de7603ce75c1eed3c04cec357